### DISCUSSION TITLE
error while trying to connect to planetscale server

### DISCUSSION THREAD

**Discussion by Nithur-M** [sidorares/node-mysql2#1868] [Q&A] [answered]
I am trying to connect to MySQL server managed by planetscale and running into an SSL issue.
`server does not allow insecure connections, client must use SSL/TLS`
Please help me resolve it. Thanks.

**Comment by sidorares** [comment, marked as answer]
you need to set `ssl: {"rejectUnauthorized":true}` in connection options ( or use a connection string the way planetscale suggests for node.js example when you generate a password )

**Reply by Nithur-M** [reply]
Thank you @sidorares, that's what I went with. But I am wondering if there will be any downsides. Thank you.

**Reply by sidorares** [reply]
You mean downsides of having ssl encryption? With planetscale you have no choice, but if you connect to your server in your own vpc it might be better to avoid ssl for perf reasons, it adds few roundtrips to a connection time

**Reply by Nithur-M** [reply]
Thanks @sidorares, May I know what `ssl: {"rejectUnauthorized":true}` is exactly doing?

**Reply by sidorares** [reply]
its passed down to tls client, see docs at https://nodejs.org/dist/latest-v18.x/docs/api/tls.html#tlsconnectoptions-callback.
